components/menus/menu-item/menu-item.html

```diff
@@ -0,0 +1,38 @@
+{# @select
+#
+# parameters:
+# id            - string, required
+# name          - string, required
+# label         - string, required
+# icon          - string
+# url           - string
+# class         - string
+# required      - boolean
+# disabled      - boolean
+# selected      - boolean
+#}
+<section class="mzp-c-menu-item {% if icon %}mzp-has-icon{% endif %} {% if class %}{{ class }}{% endif %}">
+  {% if url %}<a class="mzp-c-menu-item-link" href="{{url}}">{% else %}<div class="mzp-c-menu-item-head">{% endif %}
+    {% if icon %}
+      {% block icon %}
```

Comment:
Although Nunjucks supports blocks (same as Jinja), I'm not sure Fractal supports them in components, or at least it didn't work when I tried it (I might have done something wrong). So this might be a shortcoming we'll have to work around. Just a heads-up.

Reply:
I was wrong! Fractal does support blocks, you just have to be sure you tell the child template what parent template it's extending, e.g. `{% extends '@menu-item' %}` at the top. But this doesn't seem compatible with Fractal's `render` helper so it may not be very useful to us in the end.